Automattic/jetpack
# Upgrade Nudges: Fix in the full site editor

Fixes https://github.com/Automattic/jetpack/issues/19539

#### Changes proposed in this Pull Request:

* Make sure plan upgrade nudges work and redirect back to the correct location in the site editor.

<img width="681" alt="Screen Shot 2021-04-28 at 10 56 23 AM" src="https://user-images.githubusercontent.com/1464705/116450680-6dcc9800-a810-11eb-9cd7-41fe7dc7eb0c.png">

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Sync this branch to your sandbox on a site with a free plan.
* Open the site editor and insert a premium block.
* Make some edits to different template parts, insert a few other blocks. Make a note of the changes you made.
* Select the premium block, and hit the "Upgrade" button.
* Confirm that the "Save" button changes to a saving state, before you are redirected to a checkout page.
* Complete the checkout and confirm you are returned to the site editor in the same state.
* Confirm that all of your changes before you were redirected to checkout are still present.
* Open the post editor and confirm that the upgrade flow still works correctly here.

Note: In the site editor you will not get a notice that you have successfully purchased a plan on redirect. The site editor does not currently support the top notices. Support will be added via https://github.com/WordPress/gutenberg/pull/31303
